syntax="proto3";
package eos.console;

message QoSProto {
  message IdentifierProto {
    oneof Identifier {
      uint64 FileId = 1;
      uint64 ContainerId = 2;
      string Path = 3;
    }
  }

  message ListProto {
    string classname = 1;
  }

  message GetProto {
    IdentifierProto identifier = 1;
    repeated string key = 2;
  }

  message SetProto {
    IdentifierProto identifier = 1;
    string classname = 2;
  }

  oneof subcmd {
    ListProto list = 1;
    GetProto get = 2;
    SetProto set = 3;
  }
}
